readme.md

```diff
@@ -97,6 +97,36 @@ config.set('foo', '1');
 
 **Note:** The `default` value will be overwritten by the `defaults` option if set.
 
+### migrations
+
+Type: `Object`
+
+You can use migrations to perform operations to the store whenever a version is switched.
+
+The `migrations` object should be consisted of a key-value pair of `version`: `handler`.
+
+**Note**: The `projectVersion` option should be specified in order for the migrations to be run.
+But you only need to specify the `projectVersion` if you don't have a package.json file in your project.
```

Comment:
Perhaps remove this line which is similar to what is described in the option itself and link `projectVersion` to the option itself in the sentence above (The **`projectVersion`** option...)?

Reply:
Done, seems better now?

Reply:
Yep, but I think you have missed documenting that handlers must be idempotent and briefly explain why (they will run more than once if migrations fail).

Reply:
> > Is it possible to use, for example, store.set('**internal**.version', '1.2.3') ?
> 
> Do you mean for users? Then yes, but it should not be allowed. We should document that the `__internal__` key is reserved and throw if it's used from the outside. I doubt anyone would ever encounter it though.

I mean for internal usage. If the user has set `accessPropertiesByDotNotation` to false, the internal migration code would fail, right?

Reply:
Do you want to wait for the discussion in https://github.com/sindresorhus/conf/pull/83#discussion_r306199633 to proceed before I apply the change you suggested?

Reply:
Lets wait for @sindresorhus feedback. Still, there are situations in which the function might run twice, even if we make the effort of prevent most of them. So the comment still makes sense to be added here.